modules/core/src/lib/effect.ts

```diff
@@ -13,7 +13,7 @@ export interface Effect {
   props: any;
   useInPicking?: boolean;
 
-  preRender: (gl: WebGLRenderingContext, opts: PreRenderOptions) => void;
+  preRender: (gl: WebGLRenderingContext, opts: PreRenderOptions) => any;
```

Comment:
Nit: avoid any as far as possible
```suggestion
  preRender: (gl: WebGLRenderingContext, opts: PreRenderOptions) => unknown;
```